app.rb

```diff
@@ -0,0 +1,117 @@
+# file: app.rb
+
+require 'sinatra'
+require "sinatra/reloader"
+require 'bcrypt'
+require_relative 'lib/database_connection'
+require_relative 'lib/user_repository'
+require_relative 'lib/message_repository'
+
+DatabaseConnection.connect
+
+class Application < Sinatra::Base
+
+  enable :sessions
+
+  configure :development do
+    register Sinatra::Reloader
+    also_reload 'lib/user_repository'
+    also_reload 'lib/message_repository'
+  end
+
+  get '/' do
+    repo = MessageRepository.new
+    @messages = repo.all.sort_by(&:created_at).reverse
+
+    @tags = []
+    
+    user_info = UserRepository.new
+    tag = user_info.all
+    tag.each do | username |
+        @tags << "@#{username.username}"
+    end
+    return erb(:index)
+  end
+
+  post '/' do
+    if invalid_request_parameters?
+      status 400
+      return ''
+    end
+
+    redirect '/login' if session[:user_id] == nil
+    
+    title = params[:title]
+    content = params[:content]
+    tags = params[:tags]
```

Comment:
as per comment below, I think these variables dont get used and these three lines could be deleted?